src/System.Runtime.Extensions/src/System/IO/TextWriter.cs

```diff
@@ -161,6 +162,18 @@ public virtual void Write(char[] buffer, int index, int count)
             for (int i = 0; i < count; i++) Write(buffer[index + i]);
         }
 
+        // Writes a span of characters to the text stream.
+        //
+        public virtual void Write(ReadOnlySpan<char> source)
+        {
+            char[] buffer = ArrayPool<char>.Shared.Rent(source.Length);
+
+            source.CopyTo(new Span<char>(buffer, 0, source.Length));
+            Write(buffer, 0, source.Length);
+
+            ArrayPool<char>.Shared.Return(buffer);
```

Comment:
did you intentionally not use try/finally here and below?

Reply:
Yup I intentionally dropped try / finally here (and below) with the thought that creation of the span and `CopyTo` methods should not throw. But I now realize Write could also throw (say closed stream case)... putting the try / finally back in again...